test/integration/component/test_tags.py

```diff
@@ -1756,6 +1756,7 @@ def test_15_project_tag(self):
             listall=True,
             resourceType='project',
             resourceIds=project.id,
+            projectid=project.id,
```

Comment:
I would have expected this to work without specifying project ID. Do you think we need to fix mgmt server instead?

Reply:
@DaanHoogland Read this https://github.com/apache/cloudstack/pull/3323/files, how about we add a check to read the resource type and use the provided resource ID as account ID when type is project.

Reply:
@rhtyd that should be good.

Reply:
yes, that would work, @rhtyd. I don't think it matters much, as the id is already in the API, but that solution would have less breakage of ecosystem scripting so let's go for it.

Reply:
yes @DaanHoogland, I'm more concerned with existing systems and integrations that may break. And moreover, it breaks the API semantics (we're already passing projectid in the resource ID so why not use it).

Reply:
on second thought @rhtyd , it won't work. There is no resource ID unless the project ID is passed as such. in the call there is no information that we can use
```
            listall=True,
            resourceType='project',
            key=tag_key,
```
We could get the current account from the call context but that would only satisfy some use-cases. if an admin wants tags for some project they are not in we still need the parameter.

Reply:
never mind, i think I was stupid just there ^^. I'll revisit.

Reply:
I've been mostly busy on other things today but gave it a second look. I think it can be fixed as you suggest in the query service. I will give it a swing soon. I'll remove the java change for now but leave the test in.